ios/Plugin/NewRelicCapacitorPluginPlugin.swift

```diff
@@ -156,16 +156,18 @@ public class NewRelicCapacitorPluginPlugin: CAPPlugin {
         NewRelic.setPlatform(NRMAApplicationPlatform.platform_Capacitor)
         let selector = NSSelectorFromString("setPlatformVersion:")
         NewRelic.perform(selector, with:"1.2.1")
-        
-        if collectorAddress == nil && crashCollectorAddress == nil {
-            NewRelic.start(withApplicationToken: appKey)
-        } else {
-            NewRelic.start(withApplicationToken: appKey,
-                           andCollectorAddress: collectorAddress ?? "mobile-collector.newrelic.com",
-                           andCrashCollectorAddress: crashCollectorAddress ?? "mobile-crash.newrelic.com")
+
+        DispatchQueue.main.async {
+            if collectorAddress == nil && crashCollectorAddress == nil {
+                NewRelic.start(withApplicationToken: appKey)
+            } else {
+                NewRelic.start(withApplicationToken: appKey,
+                               andCollectorAddress: collectorAddress ?? "mobile-collector.newrelic.com",
+                               andCrashCollectorAddress: crashCollectorAddress ?? "mobile-crash.newrelic.com")
+            }
+
+            call.resolve()
```

Comment:
I moved the `resolve` call inside the main thread in order to know that we have really completed the whole job of starting up New Relic before telling Capacitor that we are done. This seems safer to me.

On the Capacitor side there doesn't seem to be any requirements on when to call `resolve` in their [official docs](https://capacitorjs.com/docs/plugins/ios#returning-data-back).  In particular they do not specify if `resolve` must be synchronous with the calling context.  NR initialization seems to work fine with `resolve` being inside the main thread closure or outside of it, so if there's a preference please let me know the reason and I can move it.

Reply:
If we are returning data from the main thread, we must call resolve from the main thread. However, you may call resolve from the outside as well without any issues.